src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -657,7 +657,7 @@ public static void Fill<T>(T[] array, T value, int startIndex, int count)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
             }
 
-            if (startIndex < 0 || startIndex > array.Length)
+            if ((uint)startIndex > (uint)array.Length)
```

Comment:
@benaadams um.. isn't it should be `>=`  here ?
It's currently incorrect (even before your PR)

Reply:
No, >= fails for `Fill(empty, default, 0, 0)` (Or any `Fill(arr, default, arr.Length, 0)`